### PR TITLE
FIX: Remove use_annotation_cache_keys params

### PR DESCRIPTION
**Pull request by maxulysse**

### CODE REVIEW

**Review by asp8200 — CHANGES_REQUESTED** [review]
Unfortunately, still doesn't work for me, see posts on Slack:

https://nfcore.slack.com/archives/C05V9FRJYMV/p1698424331245949?thread_ts=1698416130.722429&cid=C05V9FRJYMV

**Comment by asp8200** [comment]
The following cmd fails because of the hardcoding of the Sarek version number to 3.3.0

https://github.com/maxulysse/nf-core_sarek/blob/d65ecf662ef815aa5c9b31d6212637703c33be62/docs/usage.md?plain=1#L991

I get `Please specify either `--download_cache` or `--snpeff_cache`, `--vep_cache`.`

However, I get the same error when running the command from the docs on your latest commit:
```
nextflow run main.nf  --outdir results --outdir_cache /home/ubuntu/test_cache --tools vep,snpeff --download_cache --build_only_index --input false
```
The command in the docs should probably also include `--vep_cache=false` and `--snpeff_cache=false`.

File: subworkflows/local/samplesheet_to_channel/main.nf
Comment (maxulysse) [inline subworkflows/local/samplesheet_to_channel/main.nf:1]: just replaced tabs with spaces

File: nextflow_schema.json [hunk 1]
```diff
@@ -797,7 +791,7 @@
                     "type": "boolean",
                     "fa_icon": "fas fa-download",
                     "description": "Download annotation cache.",
-                    "help_text": "Set this parameter, if you wish to download annotation cache."
+                    "help_text": "Set this parameter, if you wish to download annotation cache.\nUsing this parameter will download cache even if --snpeff_cache and --vep_cache are provided."
```
Comment (asp8200) [inline nextflow_schema.json:794]: I don't understand that. In what scenario would the user want to specify `--download_cache` along with `--snpeff_cache` and `--vep_cache`?  Doesn't it just indicate that the user doesn't know what he is doing?

File: subworkflows/local/initialize_annotation_cache/main.nf [hunk 2]
```diff
@@ -0,0 +1,57 @@
+//
+// INITIALIZE ANNOTATION CACHE
+//
+
+// Initialize channels based on params or indices that were just built
+// For all modules here:
+// A when clause condition is defined in the conf/modules.config to determine if the module should be run
+// Condition is based on params.step and params.tools
+// If and extra condition exists, it's specified in comments
```
Comment (asp8200) [inline subworkflows/local/initialize_annotation_cache/main.nf:9]:
```suggestion
// If an extra condition exists, it's specified in comments
```

File: workflows/sarek.nf [hunk 3]
```diff
@@ -294,17 +253,32 @@ workflow SAREK {
...
-    snpeff_info     = params.snpeff_cache ? [] : Channel.of([ [ id:"${params.snpeff_genome}.${params.snpeff_db}" ], params.snpeff_genome, params.snpeff_db ])
-
+    // Download cache
     if (params.download_cache) {
-        PREPARE_CACHE(ensemblvep_info, snpeff_info)
-        snpeff_cache = PREPARE_CACHE.out.snpeff_cache
-        vep_cache    = PREPARE_CACHE.out.ensemblvep_cache.map{ meta, cache -> [ cache ] }
-
-        versions = versions.mix(PREPARE_CACHE.out.versions)
+        // Assuming that even if the cache is provided, if the user specify download_cache, sarek will download the cache
```
Comment (asp8200) [inline workflows/sarek.nf:258]: I would stop that pipeline with a message telling the user that the combination of options don't make sense.

**Comment by asp8200** [comment]
In`usage.md`, there are [commands](https://github.com/maxulysse/nf-core_sarek/blob/dac4b79a8ba0f16e99a3e80c2322a90c49e8a089/docs/usage.md?plain=1#L990-L998) like
```
nextflow run nf-core/sarek -r 3.3.0 --outdir results --outdir_cache /path_to/my-own-cache --tools vep,snpeff --download_cache --build_only_index --input false
```
where the version of Sarek is hardcoded to 3.3.0. Could one instead specify `-r master`? If not, then I suggest replacing `r 3.3.0` with the version number of the next release of Sarek.

There is another command there with `-r 3.3.0` which probably also needs to be updated.

By the way, I needed to add `-profile docker` to actually run the commands.

File: nextflow_schema.json (hunk 1 above)
Comment (maxulysse) [reply nextflow_schema.json:794]: These params are specified by default, so it's most likely that a user will not notice that and try to download cache which will fail

File: workflows/sarek.nf (hunk 3 above)
Comment (maxulysse) [reply workflows/sarek.nf:258]: That's what we were doing, and what's causing the issue. Given that these params have default, I think it's easier for the user to just use download_cache without having to disable 2 params if the need arise

**Comment by maxulysse** [comment]
> In`usage.md`, there are [commands](https://github.com/maxulysse/nf-core_sarek/blob/dac4b79a8ba0f16e99a3e80c2322a90c49e8a089/docs/usage.md?plain=1#L990-L998) like
> ```
> nextflow run nf-core/sarek -r 3.3.0 --outdir results --outdir_cache /path_to/my-own-cache --tools vep,snpeff --download_cache --build_only_index --input false
> ```
> where the version of Sarek is hardcoded to 3.3.0. Could one instead specify `-r master`? If not, then I suggest replacing `r 3.3.0` with the version number of the next release of Sarek.
> 
> There is another command there with `-r 3.3.0` which probably also needs to be updated.
> 
> By the way, I needed to add `-profile docker` to actually run the commands.
>  

Good point about the version, I'll update that.

For the profile, I'd say it's implied given the previous instructions that you have to use one

File: docs/usage.md [hunk 4]
```diff
@@ -44,7 +50,7 @@ Do not use `-c <file>` to specify parameters as this will result in errors. Cust
 The above pipeline run specified with a params file in yaml format:
 
 ```bash
-nextflow run nf-core/sarek -profile docker -params-file params.yaml
+nextflow run nf-core/sarek -params-file params.yaml
```
Comment (FriederikeHanssen) [inline docs/usage.md:53]: why are we removing docker here?

**Review by FriederikeHanssen — COMMENTED** [review]
looks good. but why remove `-profile`?

File: docs/usage.md [hunk 5]
```diff
@@ -962,7 +962,7 @@ aws s3 --no-sign-request ls s3://annotation-cache/vep_cache/
 
 Since both Snpeff and VEP are internally figuring the path towards the specific cache version / species, `annotation-cache` is using an extra set of keys to specify the species and genome build.
 
-So if you are using this resource, please either set `--use_annotation_cache_keys` to use the AWS annotation cache, or point towards your own cache folder structure matching the expected structure.
+Which is handled internally by Sarek.
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply docs/usage.md:971]: > `annotation-cache` is using an extra set of keys to specify the species and genome build.

**Comment by maxulysse** [comment]
> looks good. but why remove `-profile`?

I felt it was a mess because we have that in some command but not others, so I'd rather remove it everywhere and tell people to use the profile they need

File: subworkflows/local/initialize_annotation_cache/main.nf (hunk 2 above)
Comment (maxulysse) [reply subworkflows/local/initialize_annotation_cache/main.nf:9]: not fixing that here, as it's a copy paste and I'm sure it's in every other subworkflows, I'll make a separate PR for it

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**